linera-service/src/linera-exporter/runloops/block_processor.rs

```diff
@@ -0,0 +1,461 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{collections::HashSet, future::IntoFuture, sync::Arc, time::Duration};
+
+use linera_base::crypto::CryptoHash;
+use linera_chain::types::{Block, ConfirmedBlockCertificate};
+use linera_storage::Storage;
+use tokio::{
+    sync::mpsc::{UnboundedReceiver, UnboundedSender},
+    time::{interval, MissedTickBehavior},
+};
+
+use crate::{
+    common::{BlockId, ExporterError, LiteBlockId},
+    storage::BlockProcessorStorage,
+};
+
+pub(super) struct BlockProcessor<T>
+where
+    T: Storage + Clone + Send + Sync + 'static,
+{
+    storage: BlockProcessorStorage<T>,
+    queue_rear: UnboundedSender<BlockId>,
+    queue_front: UnboundedReceiver<BlockId>,
+}
+
+struct Walker<'a, S>
+where
+    S: Storage + Clone + Send + Sync + 'static,
+{
+    path: Vec<NodeVisitor>,
+    visited: HashSet<BlockId>,
+    storage: &'a mut BlockProcessorStorage<S>,
+}
+
+#[derive(Debug)]
+struct ProcessedBlock {
+    block: BlockId,
+    siblings: Vec<BlockId>,
```

Comment:
I agree with Adreas here. I think _siblings_ isn't the best name. Since the only thing it ever is now are blocks that sent the message we included in the current block - let's rename to `sender_blocks` and add a comment.